src/engraving/libmscore/textedit.cpp

```diff
@@ -222,6 +222,26 @@ void TextBase::endEdit(EditData& ed)
     } else {
         triggerLayout();
     }
+    if (isLyrics()) {
```

Comment:
Why is this code here (in the base class) and not inside Lyrics?

Reply:
The idea behind this section is that when a lyric is finished editing, it checks to see if there is a previous lyric with a line that extends through the current lyric's tick, and if so, resizes the previous lyric's line (or removes it) so that there is no overlap.

Reply:
> We could move this, but I'm not sure the best place to do it

This code looks like it should be inside the Lyrics class (since it's a lyrics-specific logic and TextBase is the base class for all text elements)

Reply:
I still elected to call it from the base class, because the base class `endEdit` finishes the undo command, and it also has some early return logic that I felt would be a giant waste to copy and paste into the lyrics class. If this is still not good enough, I can move it into the lyrics class and replicate the early return checks from the base class.